Library/DasharoSystemFeaturesUiLib/DasharoSystemFeatures.c

```diff
@@ -457,6 +462,49 @@ DasharoSystemFeaturesUiLibConstructor (
         );
     ASSERT_EFI_ERROR (Status);
   }
+
+  BufferSize = sizeof (mDasharoSystemFeaturesPrivate.DasharoFeaturesData.DmaProtection);
+  Status = gRT->GetVariable (
+      mDmaProtectionEfiVar,
+      &gDasharoSystemFeaturesGuid,
+      NULL,
+      &BufferSize,
+      &mDasharoSystemFeaturesPrivate.DasharoFeaturesData.DmaProtection
+      );
+
+  if (Status == EFI_NOT_FOUND) {
+    Status = gRT->SetVariable (
+        mDmaProtectionEfiVar,
+        &gDasharoSystemFeaturesGuid,
+        EFI_VARIABLE_BOOTSERVICE_ACCESS | EFI_VARIABLE_NON_VOLATILE,
+        sizeof (mDmaProtectionDefault),
+        &mDmaProtectionDefault
+        );
+    mDasharoSystemFeaturesPrivate.DasharoFeaturesData.DmaProtection = mDmaProtectionDefault;
+    ASSERT_EFI_ERROR (Status);
+  }
+
+  BufferSize = sizeof (mDasharoSystemFeaturesPrivate.DasharoFeaturesData.IommuHandoff);
+  Status = gRT->GetVariable (
+      mIommuHandoffEfiVar,

```

Comment:
The handoff may be under the same variable, e.g. make it:

```
struct IommuConfiguration {
  BOOLEAN DmaProtectionEnable;
  UINT8        IommuHandoff;
};
```

Then we save all the hassle with setting the variables for each option and reduce the overhead in the code and in the flash.

Reply:
Like I said in a comment below, would like to avoid changes like these since this PR spans over 3 repositories.

Reply:
And the software here will span into 30 thousands lines of code in no time if we keep such an attitude. Please apply my suggestion

Reply:
I've changed it to a single variable. I've used one UINT8 and defined macros to disable/enable dma protection in the mode that we want. I've tried doing this with a struct, but couldn't find a way to switch it with `oneof` in the `.vfr ` file, without adding another useless Efi variable. This way the code ended up being even shorter so I hope it's fine now.